src/user.cpp

```diff
@@ -99,11 +126,23 @@ User* User::unserialize(MegaClient* client, string* d)
     l = *ptr++;
     if (l)
     {
+        if (ptr + l > end)
+        {
+            return NULL;
```

Comment:
It would be good to have warning or error logs in these cases to detect problems serializing/unserializing objects